stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/querychannels/NonMemberChatEventHandler.kt

```diff
@@ -0,0 +1,108 @@
+package io.getstream.chat.android.offline.querychannels
+
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.api.models.FilterObject
+import io.getstream.chat.android.client.events.ChannelUpdatedByUserEvent
+import io.getstream.chat.android.client.events.ChannelUpdatedEvent
+import io.getstream.chat.android.client.events.MemberAddedEvent
+import io.getstream.chat.android.client.events.MemberRemovedEvent
+import io.getstream.chat.android.client.events.NotificationAddedToChannelEvent
+import io.getstream.chat.android.client.events.NotificationMessageNewEvent
+import io.getstream.chat.android.client.events.NotificationRemovedFromChannelEvent
+import io.getstream.chat.android.client.models.Channel
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.runBlocking
+
+/**
+ * Implementation of [ChatEventHandler] that handles events when ChannelListViewModel has a filter to show channels which
+ * the current user is not a member. BE AWARE that this implementation uses much more API calls than [DefaultChatEventHandler]
+ */
+public class NonMemberChatEventHandler(
+    private val client: ChatClient,
+    private val channels: StateFlow<List<Channel>>,
+) : BaseChatEventHandler() {
+
+    override fun handleNotificationAddedToChannelEvent(
+        event: NotificationAddedToChannelEvent,
+        filter: FilterObject,
+    ): EventHandlingResult = EventHandlingResult.Skip
+
+    override fun handleMemberAddedEvent(
+        event: MemberAddedEvent,
+        filter: FilterObject,
+    ): EventHandlingResult = handleMemberUpdate(event.cid, filter)
+
+    override fun handleChannelUpdatedByUserEvent(
+        event: ChannelUpdatedByUserEvent,
+        filter: FilterObject,
+    ): EventHandlingResult = EventHandlingResult.Skip
+
+    override fun handleChannelUpdatedEvent(event: ChannelUpdatedEvent, filter: FilterObject): EventHandlingResult =
+        EventHandlingResult.Skip
+
+    /**
+     * Handles [NotificationMessageNewEvent]. It makes a request to API to define outcome of handling.
+     *
+     * @param event Instance of [NotificationMessageNewEvent] that is being handled.
+     * @param filter [FilterObject] which is used to define an outcome.
+     */
+    override fun handleNotificationMessageNewEvent(
+        event: NotificationMessageNewEvent,
+        filter: FilterObject,
+    ): EventHandlingResult = EventHandlingResult.Skip
+
+    /**
+     * Handles [MemberRemovedEvent]. It makes a request to API to define outcome of handling.
+     *
+     * @param event Instance of [MemberRemovedEvent] that is being handled.
+     * @param filter [FilterObject] which is used to define an outcome.
+     */
+    override fun handleMemberRemovedEvent(
+        event: MemberRemovedEvent,
+        filter: FilterObject,
+    ): EventHandlingResult = handleMemberUpdate(event.cid, filter)
```

Comment:
I think we don't have to do this request if we are assuming that channel should be visible event if you are not the member

Reply:
We need to add the channel because the user is no longer a member. If the user was previously there, it need to be removed.